ff/src/fields/field_hashers/expander/mod.rs

```diff
@@ -2,98 +2,114 @@
 // With some optimisations
 
 use ark_std::vec::Vec;
-use digest::{DynDigest, ExtendableOutput, Update};
+
+use arrayvec::ArrayVec;
+use digest::{ExtendableOutput, FixedOutputReset, Update};
+
 pub trait Expander {
-    fn construct_dst_prime(&self) -> Vec<u8>;
     fn expand(&self, msg: &[u8], length: usize) -> Vec<u8>;
 }
 const MAX_DST_LENGTH: usize = 255;
 
-const LONG_DST_PREFIX: [u8; 17] = [
-    //'H', '2', 'C', '-', 'O', 'V', 'E', 'R', 'S', 'I', 'Z', 'E', '-', 'D', 'S', 'T', '-',
-    0x48, 0x32, 0x43, 0x2d, 0x4f, 0x56, 0x45, 0x52, 0x53, 0x49, 0x5a, 0x45, 0x2d, 0x44, 0x53, 0x54,
-    0x2d,
-];
+const LONG_DST_PREFIX: &[u8; 17] = b"H2C-OVERSIZE-DST-";
+
+/// Implements section [5.3.3](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#section-5.3.3)
+/// "Using DSTs longer than 255 bytes" of the
+/// [IRTF CFRG hash-to-curve draft #16](https://datatracker.ietf.org/doc/html/draft-irtf-cfrg-hash-to-curve-16#section-5.3.3).
+pub struct DST(arrayvec::ArrayVec<u8, MAX_DST_LENGTH>);
+
+impl DST {
+    pub fn new_xmd<H: FixedOutputReset + Default>(dst: &[u8]) -> DST {
+        DST(if dst.len() > MAX_DST_LENGTH {
+            let mut long = H::default();
+            long.update(&LONG_DST_PREFIX[..]);
+            long.update(&dst);
+            ArrayVec::try_from(long.finalize_fixed().as_ref()).unwrap()
+        } else {
+            ArrayVec::try_from(dst).unwrap()
+        })
```

Comment:
Let's do something like 
```suggestion
        let array = if dst.len() > MAX_DST_LENGTH {
            let mut long = H::default();
            long.update(&LONG_DST_PREFIX[..]);
            long.update(&dst);
            (long.finalize_fixed().as_ref()).try_into()
        } else {
            dst.try_into()
        };
        DST(array.unwrap())
```